### PR TITLE
Add a pure C++ version for comparison

### PR DESCRIPTION
**Pull request by terryzhao127**
This version runs purely on CPU, which shows the improvements achieved by CUDA.

### CODE REVIEW

**Comment by jiangyangzhou** [comment]
Wow interesting. Thanks for notice this repo, have you tested the performance yet?  This cuda implement is very fast (Easily run 3000 times for 4^2 depth search)
Sorry for miss the commit for 3 months, I'll test this request as soon as possible.

File: mt_2048.cpp [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    int count = 0;
+    while (zero_num != -1) {
+        count += 1;
+        turn = rand() % 4;
+        result = combine(l_state, turn);
+        score += result * (result > 0);
+        zero_num = isOver(l_state);
+        if (zero_num > 0 && result != -1) newBlock(l_state);
+    }
+    d_result[tid] = score;
```
Comment (jiangyangzhou) [inline mt_2048.cpp:222]: I think here should be d_result[tid += score. Just like “atotic_add” in my .cu implement

**Review by jiangyangzhou — CHANGES_REQUESTED** [review]
I think in line 222 should be
` d_result[tid] += score`

File: mt_2048.cpp (hunk 1 above)
Comment (terryzhao127) [reply mt_2048.cpp:222]: You are right! :laughing:

**Review by terryzhao127 — COMMENTED** [review]
The scores should be summed up, which must be corrected.

**Comment by jiangyangzhou** [comment]
Hi I add performance analysis, .cu program is the fastest one!

**Comment by terryzhao127** [comment]
Cool! This is the decisive proof that the CUDA version has higher performance~

**Pull request merged**